finalfrontier/src/config.rs

```diff
@@ -128,6 +128,36 @@ pub struct SubwordVocabConfig {
     pub discard_threshold: f32,
 }
 
+/// Hyperparameters for subword-sized-vocabs.
+#[derive(Clone, Copy, Debug, Serialize)]
+#[serde(rename = "SubwordSizedVocab")]
+#[serde(tag = "type")]
+pub struct SubwordSizedVocabConfig {
```

Comment:
I am wondering if we need new structs for this, they are largely overlapping. If we add some other field, we have to add them to 2 or 4 structs. Maybe replace `vocab_size`/`mincount` counts an enum along the lines of:

```rust
pub enum VocabCutoff {
    MinCount(usize),
    VocabSize(usize),
}
```

Reply:
I'm in favour of this, although `VocabSize` could be misinterpreted if someone prints the metadata. Perhaps `TargetSize`?

Reply:
Good point, I propose an extension to your suggestion: `TargetVocabSize`

Reply:
I was thinking the same. I like the idea of `TargetVocabSize`.